draft-cpbs-pseudorandom-ctls.md

```diff
@@ -182,30 +183,74 @@ Servers SHOULD expand any Alert message following the ClientHello to the same si
 
+# The `TLS_AES_(128/256)_SHA256_SIV` Cipher Suites {#aes-sha2-siv}
+
+The Pseudorandom cTLS extension is sufficient to render the bitstream pseudorandom to a third party when both peers are operating correctly.  However, if a malicious client or server can coerce its peer into sending particular plaintext (as is common in web browsers), it can choose plaintext with knowledge of the encryption keys, in order to produce ciphertext that has visible structure to a third party.  This technique can be used to mount protocol confusion attacks {{SLIPSTREAM}}.
+
+This attack is particularly straightforward when using the AES-GCM or ChaCha20-Poly1305 cipher suites, as much of the ciphertext is encrypted by XOR with a stream cipher.  A malicious peer in this threat model can choose desired ciphertext, encrypt it with that keystream to produce the plaintext, and rely on the other peer's encryption stage to reverse the encryption and reveal the desired ciphertext.
+
+As a defense for this threat model, this draft introduces the `TLS_AES_(128/256)_SHA256_SIV` cipher suites.  These cipher suites provide an AEAD algorithm {{!RFC5116}} with `K_LEN = 16` or `32`, `N_MIN = 0`, `N_MAX = 255`, and `A_MAX = P_MAX = infinity`.  They employ a Synthetic Initialization Vector construction, similar to SIV-AES {{?RFC5297}} and AES-GCM-SIV {{?RFC8452}} but using HMAC-SHA256 {{!RFC2104}} as the MAC.  We call this AEAD family "AES-SHA2-SIV".
+
+In SIV cipher modes, the MAC output initializes the encryption process.  In AES-SHA2-SIV, the MAC is a secure hash function even when the key is known.  The sender therefore cannot control any bit of the ciphertext except by trial encryption.  Fixing `N` bits of the ciphertext to desired values requires the attacker to perform `2^N` trial encryptions, so fixing more than 128 bits of ciphertext to desired values is computationally infeasible.  These trials cannot begin until after the TLS handshake, so practical limits on `N` are likely to be considerably lower.
+
+In formal terms, this construction prevents known-key distinguishing attacks {{?KNOWNKEY=DOI.10.1007/978-3-662-43933-3_18}}.  AES-SHA2-SIV is also nonce-misuse-resistant, although this is not relevant to TLS.
+
+In these cipher suites, HMAC-SHA256 is also used to construct the HKDF for the TLS handshake.
+
+These cipher suites are less efficient than AES-GCM, so they SHOULD NOT be used unless Pseudorandom cTLS is enabled and ciphertext confusion attacks are relevant.  Their computational cost is expected to be similar to the `TLS_*_WITH_AES_(128/256)_CBC_SHA256` cipher suites from TLS 1.2 ({{Appendix A.5 of ?RFC5246}}).  Encryption requires two passes over each message, but decryption can still be performed in a single pass.
+
+## Encryption
+
```

Comment:
This is indeed A4 from [[NRS14](https://eprint.iacr.org/2014/206.pdf)]. Can we add a reference to that paper here?

Reply:
Done (in an an earlier paragraph).